### PR TITLE
Add Gradient Clip

### PR DESCRIPTION
**Pull request by jalencato**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/trainer/ep_trainer.py [hunk 1]
```diff
@@ -152,6 +156,8 @@ def fit(self, train_loader, num_epochs,
                 self.optimizer.step()
                 rt_profiler.record('train_step')
 
+                if max_grad_norm is not None:
+                    th.nn.utils.clip_grad_norm_(model.parameters(), max_grad_norm)
```
[the remark this answers is not in the document]
Comment (jalencato) [reply python/graphstorm/trainer/ep_trainer.py:160]: I think we can provide an option to add norm_type as well. Will do it asap.

File: tests/end2end-tests/graphstorm-lp/test.sh [hunk 2]
```diff
@@ -77,6 +77,11 @@ python3 -m graphstorm.run.gs_link_prediction --workspace $GS_HOME/training_scrip
 
 error_and_exit $?
 
+echo "**************dataset: Movielens, RGCN layer 1, node feat: fixed HF BERT, BERT nodes: movie, inference: mini-batch, negative_sampler: uniform, exclude_training_targets: true, gradient clip: 0.1, gradient norm type: 1"
+python3 -m graphstorm.run.gs_link_prediction --workspace $GS_HOME/training_scripts/gsgnn_lp --num-trainers $NUM_TRAINERS --num-servers 1 --num-samplers 0 --part-config /data/movielen_100k_lp_train_val_1p_4t/movie-lens-100k.json --ip-config ip_list.txt --ssh-port 2222 --cf ml_lp.yaml --train-negative-sampler uniform --exclude-training-targets True --reverse-edge-types-map user,rating,rating-rev,movie --num-epochs 1 --eval-frequency 300 --max-grad-norm 0.1 --grad-norm-type 1
```
Comment (GentleZhu) [inline tests/end2end-tests/graphstorm-lp/test.sh:81]: grad-norm-type default should be 2.0 ? Why it is different graphstorm-nc/test.sh

Comment (jalencato) [reply tests/end2end-tests/graphstorm-lp/test.sh:81]: I suppose user wants to use L2-norm gradient clip as default as it is popular, for the graphstorm-nc/test.sh, it is used as an end-to-end test. So I want to use different setting as the graphstorm-lp.

**Review by GentleZhu — APPROVED** [review]
